src/test/java/com/squareup/kotlinpoet/CrossplatformTest.kt

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (C) 2017 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.squareup.kotlinpoet
+
+import com.google.common.truth.Truth.assertThat
+import org.junit.Test
+import java.util.concurrent.atomic.AtomicReference
+
+class CrossplatformTest {
+
+  @Test fun crossplatform() {
+    val headerTypeParam = TypeVariableName("V")
+    val headerType = "AtomicRef"
+    val headerSpec = TypeSpec.headerClassBuilder(headerType)
+        .addTypeVariable(headerTypeParam)
+        .addModifiers(KModifier.INTERNAL)
+        .primaryConstructor(FunSpec.constructorBuilder()
+            .addParameter("value", headerTypeParam)
+            .build())
+        .addProperty(PropertySpec.builder("value", headerTypeParam).build())
+        .addFunction(FunSpec.builder("get")
+            .returns(headerTypeParam)
+            .build())
+        .addFunction(FunSpec.builder("set")
+            .addParameter("value", headerTypeParam)
+            .build())
+        .addFunction(FunSpec.builder("getAndSet")
+            .addParameter("value", headerTypeParam)
+            .returns(headerTypeParam)
+            .build())
+        .addFunction(FunSpec.builder("compareAndSet")
+            .addParameter("expect", headerTypeParam)
+            .addParameter("update", headerTypeParam)
+            .returns(Boolean::class)
+            .build())
+        .build()
+    val implName = ParameterizedTypeName.get(AtomicReference::class.asTypeName(), headerTypeParam)
+    val implSpec = TypeAliasSpec.builder(headerType, implName)
+        .addTypeVariable(headerTypeParam)
+        .addModifiers(KModifier.IMPL)
+        .build()
+    val fileSpec = FileSpec.builder("", "Test")
+        .addType(headerSpec)
+        .addTypeAlias(implSpec)
+        .build()
+
+    assertThat(fileSpec.toString()).isEqualTo("""
+      |import java.util.concurrent.atomic.AtomicReference
+      |import kotlin.Boolean
+      |
+      |header internal class AtomicRef<V>(value: V) {
```

Comment:
I'm surprised the access modifier splits `header` and `class`. I think of it as one keyword!

Reply:
(though if this matches the jetbrains' order, keep it!)

Reply:
They haven't defined the order of `header` or `impl` in relation to other keywords yet.

Reply:
It would indeed be more convenient to emit `header class` together, also `internal header class` reads better IMO

Reply:
Actually, we don't want any initializers, so I guess a null-check is sufficient. Pushed cb3c5e4